src/test/java/org/nuxeo/audit/storage/impl/TestSQLAuditStorage.java

```diff
@@ -0,0 +1,189 @@
+import org.nuxeo.ecm.core.test.DefaultRepositoryInit;
+import org.nuxeo.ecm.core.test.annotations.Granularity;
+import org.nuxeo.ecm.core.test.annotations.RepositoryConfig;
+import org.nuxeo.ecm.directory.Session;
+import org.nuxeo.ecm.directory.sql.SQLDirectoryFeature;
+import org.nuxeo.ecm.platform.audit.AuditFeature;
+import org.nuxeo.ecm.platform.audit.api.AuditQueryBuilder;
+import org.nuxeo.ecm.platform.audit.api.LogEntry;
+import org.nuxeo.ecm.platform.audit.api.Predicates;
+import org.nuxeo.ecm.platform.audit.impl.LogEntryImpl;
+import org.nuxeo.ecm.platform.audit.service.NXAuditEventsService;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+/**
+ * @since 9.3
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ SQLDirectoryFeature.class, AuditFeature.class })
+@RepositoryConfig(init = DefaultRepositoryInit.class, cleanup = Granularity.METHOD)
+@Deploy({ "org.nuxeo.audit.storage" })
+public class TestSQLAuditStorage {
+
+    protected static SQLAuditStorage storage;
+
+    protected static String jsonEntry1 = "{\"entity-type\":\"logEntry\",\"category\":\"Document\",\"principalName\":\"Administrator\","
+            + "\"comment\":null,\"docLifeCycle\":\"Draft\",\"docPath\":\"/My doc 1\",\"docType\":\"File\","
+            + "\"eventId\":\"documentCreated\",\"repositoryId\":\"default\",\"eventDate\":\"2017-10-10T10:35:13.102Z\","
+            + "\"docUUID\":\"3f86a83f-1523-432a-92c5-8ec5f68a6451\",\"logDate\":\"2017-10-10T10:35:13.138Z\",\"extended\":{}}";
+
+    protected static String jsonEntry2 = "{\"entity-type\":\"logEntry\",\"category\":\"Document\",\"principalName\":\"Administrator\","
+            + "\"comment\":null,\"docLifeCycle\":\"Approved\",\"docPath\":\"/My doc 2\",\"docType\":\"File\","
+            + "\"eventId\":\"documentModified\",\"repositoryId\":\"default\",\"eventDate\":\"2017-11-13T09:15:13.102Z\","
+            + "\"docUUID\":\"4f86a82f-3521-132b-92c3-6ac5f65c6422\",\"logDate\":\"2017-11-13T09:15:13.138Z\",\"extended\":{}}";
+
+    protected static List<String> jsonEntries = Arrays.asList(jsonEntry1, jsonEntry2);
+
+    @BeforeClass
+    public static void before() {
+        NXAuditEventsService audit = (NXAuditEventsService) Framework.getRuntime()
+                                                                     .getComponent(NXAuditEventsService.NAME);
+        storage = (SQLAuditStorage) audit.getAuditStorage(SQLAuditStorage.NAME);
+    }
+
+    @Test
+    public void testStorage() {
+        assertNotNull(storage);
+        assertTrue(storage instanceof SQLAuditStorage);
+    }
+
+    @SuppressWarnings("deprecation")
+    @Test
+    public void testAppend() {
+        storage.append(jsonEntries);
+
+        Session session = storage.sqlDirectory.getSession();
+        String schemaName = storage.sqlDirectory.getSchema();
```

Comment:
Use the `getSqlDirectory()` instead of the `sqlDirectory` field.

    Directory sqlDirectory = storage.getSqlDirectory();
    Session session = sqlDirectory.getSession();
    String schemaName = sqlDirectory.getSchema();